### PR TITLE
chore(meta): bump sea-orm to 1.0

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Bump `sea-orm` to `1.0`, along with its dependencies.

The motivation is https://github.com/risingwavelabs/risingwave/pull/19156, where the necessary fix (https://github.com/SeaQL/sea-orm/pull/2244) is only included since `1.0`.

The only worth-noting breaking change is that, there is no longer a backend-agnostic method to specify `longblob` for MySQL, `blob` for SQLite, and `bytea` for Postgres simultaneously (https://github.com/SeaQL/sea-query/pull/743, a regression to me). As a workaround, add a helper method which dispatches the type based on the backend variant.

Although the latest version of `sea-orm` is `1.1`, we are not upgrading to it in this PR because it depends on a new major version of `sqlx`, for which we are maintaining our own forks to make it compatible with madsim. We may further bump it in following PRs.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
#19145  🥵

**Comment by kwannoel** [comment]
> #19145 🥵

I think we made similar changes. Shall I include your changes in my PR? I think the trait extension changes are more ergonomic.

**Comment by kwannoel** [comment]
> > #19145 🥵
> 
> I think we made similar changes. Shall I include your changes in my PR? I think the trait extension changes are more ergonomic.

I think we can also merge in your PR, then my PR will just contain `Cargo.lock` and `Cargo.toml` patches.

**Review by kwannoel — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -4489,6 +4489,18 @@ dependencies = [
  "zeroize",
 ]
 
+[[package]]
+name = "educe"
+version = "0.5.11"
```
Comment (kwannoel) [inline Cargo.lock:4494]: Just making sure the duplicated dependency here is expected.

**Review by kwannoel — APPROVED** [review]

**Review by xxchan — APPROVED** [review]
rubber stamp for Cargo.lock

**Comment by BugenZhao** [comment]
> > > #19145 🥵
> > 
> > 
> > I think we made similar changes. Shall I include your changes in my PR? I think the trait extension changes are more ergonomic.
> 
> I think we can also merge in your PR, then my PR will just contain `Cargo.lock` and `Cargo.toml` patches.

Sorry, I just noticed your work. 🥵

**Pull request merged**